docs/articles/documentation/guides/basic-guides/obtain-client-side-info.md

```diff
@@ -0,0 +1,173 @@
+---
+layout: docs
+title: Obtain Client-Side Info
+permalink: /documentation/guides/basic-guides/obtain-client-side-info.html
+---
+# Obtain Client-Side Info
+
+TestCafe allows you to create [*Client Functions*](../../reference/test-api/global/clientfunction.md) that can return any serializable value from the client side, such as the current URL or custom data calculated by a client script.
+
+> Do not modify the tested webpage within client functions.
+> Use [test actions](interact-with-the-page.md) to interact with the page instead.
+
+This topic contains the following sections.
+
+* [Client Function Constructor](#client-function-constructor)
+* [Run Asynchronous Client Code](#run-asynchronous-client-code)
+* [Execute Client Functions](#execute-client-functions)
+* [Overwrite Client Function Options](#overwrite-client-function-options)
+* [One-Time Client Code Execution](#one-time-client-code-execution)
+* [Call Selectors and Client Functions from Client-Side Code](#call-selectors-and-client-functions-from-client-side-code)
+    * [Import Functions to be Used as Client Function Dependencies](#import-functions-to-be-used-as-client-function-dependencies)
+* [Call Client Functions from Node.js Callbacks](#call-client-functions-from-nodejs-callbacks)
+* [Client Function Limitations](#client-function-limitations)
+
+## Client Function Constructor
+
+You can use the [ClientFunction](../../reference/test-api/global/clientfunction.md) constructor to create a client function.
+
+```js
+import { ClientFunction } from 'testcafe';
+const getWindowLocation = ClientFunction(() => window.location);
+```
+
+> Client functions cannot return DOM nodes. Use [selectors](select-page-elements.md) to achieve this.
+
+## Run Asynchronous Client Code
+
+TestCafe allows you to create client functions that run asynchronous code.
+
+You can pass a function that returns a Promise to the `ClientFunction` constructor to achieve this. In this instance, the client function will complete only when the Promise resolves.
+
+```js
+import { ClientFunction } from 'testcafe';
+
+const performAsyncOperation = ClientFunction(() => {
+    return new Promise(resolve => {
+        window.setTimeout(resolve, 500); // some async operations
+    });
+});
+```
+
+## Execute Client Functions
+
+Call the client function with the `await` keyword to execute it.
+
+```js
+import { ClientFunction } from 'testcafe';
+
+const getWindowLocation = ClientFunction(() => window.location);
+
+fixture `My fixture`
+    .page `http://www.example.com/`;
+
+test('My Test', async t => {
+    const location = await getWindowLocation();
+});
+```
+
+## Overwrite Client Function Option
+
+You can overwrite client function options with the ClientFunction's [with method](../../reference/test-api/clientfunction/with.md).
```

Comment:
Change to:
You can overwrite client function options via the ClientFunction's [with method]

Also, regarding "[with method]", can you make "with" either bold or italic so that it stands out from "method"?